python/lsst/daf/butler/registry/managers.py

```diff
@@ -128,7 +140,37 @@ def fromConfig(cls, config: RegistryConfig) -> RegistryManagerTypes:
         types : `RegistryManagerTypes`
             A new struct containing type objects.
         """
-        return cls(**{f.name: doImportType(config["managers", f.name]) for f in dataclasses.fields(cls)})
+        # We only check for manager names defined in class attributes.
+        # TODO: Maybe we need to check keys for unknown names/typos?
+        managers = {field.name for field in dataclasses.fields(cls)} - {"manager_configs"}
+        # Values of "config" sub-key, if any, indexed by manager name.
+        configs: Dict[str, Mapping] = {}
+        manager_types: Dict[str, Type] = {}
+        for manager in managers:
+            manager_config = config["managers"].get(manager)
+            if isinstance(manager_config, Config):
+                # Expect "cls" and optional "config" sub-keys.
+                manager_config_dict = manager_config.toDict()
+                try:
+                    class_name = manager_config_dict.pop("cls")
+                except KeyError:
+                    raise KeyError(f"'cls' key is not defined in {manager!r} manager configuration")
```

Comment:
```suggestion
                    raise KeyError(f"'cls' key is not defined in {manager!r} manager configuration") from None
```
This will avoid those noisy "while one exception was being handled..." extra tracebacks.

Reply:
👍 Old habits, I still keep forgetting that I need `from` for exceptions.